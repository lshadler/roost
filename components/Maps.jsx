import React from 'react';
import GoogleMapReact from 'google-map-react';
import Marker from './Marker';
// TODO: replace with API call?
import places from '../test-data/places';

const Maps = ({zoom}) => {
  let sumLat = 0; let sumLng = 0;

  const Markers = places.map((props) => {
    sumLat += parseFloat(props.lat);
    sumLng += parseFloat(props.lng);

    return <Marker key={props.lat + 1000 * props.lng} {...props} />;
  });

  const center = {
    lat: sumLat / places.length,
    lng: sumLng / places.length,
  };

  return (
    <div className="maps-container">
      <GoogleMapReact
        bootstrapURLKeys={{key: process.env.MAPS_API_KEY}}
        defaultCenter={center}
        defaultZoom={zoom}>
        {Markers}
      </GoogleMapReact>
      <style jsx>{`
        .maps-container {
          margin: 10px 0;
          height: 500px;
          width: 90vw;      
          box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12);
          border-radius: 4px;    
        }

        .maps-container > div > div {
          border-radius: 4px;    
        }

      `}</style>
    </div>

  );
};

Maps.defaultProps = {
  zoom: 13,
};

export default Maps;
