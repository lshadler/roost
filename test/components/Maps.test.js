import React from 'react';
import {render} from '@testing-library/react';

import Maps from '../../components/Maps';

describe('Maps Component', () => {
  test('renders in default state', () => {
    const {asFragment} = render(<Maps />);

    expect(asFragment()).toMatchSnapshot();
  });


  test('rerender to change zoom', () => {
    const {asFragment, rerender} = render(<Maps zoom={3}/>);

    expect(asFragment()).toMatchSnapshot();

    rerender(<Maps zoom={4} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
