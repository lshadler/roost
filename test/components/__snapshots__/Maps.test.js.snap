// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps Component renders in default state 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-1 MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-31"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Maps Component rerender to change zoom 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-44 MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-74"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Maps Component rerender to change zoom 2`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root WithStyles(ForwardRef(Paper))-root-44 MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-74"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
